src/advancedHMC_MCMC.jl

```diff
@@ -0,0 +1,459 @@
+                                                                          phystd = [0.05],
+                                                                          priorsNNw = (0.0,3.0))
+
+# solving ode + estimating parameters hence dataset needed to optimize parameters upon + Pior Distributions for ODE params
+fh_mcmc_chainflux2, fhsamplesflux2, fhstatsflux2 = ahmc_bayesian_pinn_ode(prob,chainflux1,
+                                                                          dataset = dataset,
+                                                                          draw_samples = 1500,
+                                                                          l2std = [0.05],
+                                                                          phystd = [0.05],
+                                                                          priorsNNw = (0.0,3.0),
+                                                                          param = [Normal(6.5,0.5),Normal(-3,0.5)])
+
+## NOTES 
+Dataset is required for accurate Parameter estimation + solving equations
+Incase you are only solving the Equations for solution, do not provide dataset
+
+## Positional Arguments
+prob -> DEProblem(out of place and the function signature should be f(u,p,t)
+chain -> Lux/Flux Neural Netork which would be made the Bayesian PINN
+dataset -> Vector containing Vectors of corresponding u,t values 
+init_params -> intial parameter values for BPINN (ideally for multiple chains different initializations preferred)
+nchains -> number of chains you want to sample (random initialisation of params by default)
+draw_samples -> number of samples to be drawn in the MCMC algorithms (warmup samples are ~2/3 of draw samples)
+l2std -> standard deviation of BPINN predicition against L2 losses/Dataset
+phystd -> standard deviation of BPINN predicition against Chosen Underlying ODE System
+priorsNNw -> Vector of [mean, std] for BPINN parameter. Weights and Biases of BPINN are Normal Distributions by default
+param -> Vector of chosen ODE parameters Distributions in case of Inverse problems.
+autodiff -> Boolean Value for choice of Derivative Backend(default is numerical)
+physdt -> Timestep for approximating ODE in it's Time domain. (1/20.0 by default)
+
+#update as AdvancedHMC has added named structs for algos
+Proposal -> Choice of MCMC Sampling Algorithm (AdvancedHMC.jl implemenations)
+targetacceptancerate -> Target percentage(in decimal) of iterations in which the proposals were accepted(0.8 by default)
+Adaptor -> https://turinglang.org/AdvancedHMC.jl/stable/
+Integrator -> https://turinglang.org/AdvancedHMC.jl/stable/
+Metric -> https://turinglang.org/AdvancedHMC.jl/stable/
+
+## References  
+
+"""
+
+mutable struct LogTargetDensity{C, S, I}
+    dim::Int
+    prob::DiffEqBase.DEProblem
+    chain::C
+    st::S
+    dataset::Vector{Vector{Float64}}
+    priors::Vector{Distribution}
```

Comment:
This is going to be unstable: does it not effect performance?

Reply:
ive been doing @time and @btime on the function on the merged struct version and the version you put this review (but uses prob::ODEProblem) the difference dosent change much. In fact on many runs the unstable version performs better.